python/tests/test_tree_stats.py

```diff
@@ -3604,23 +3759,29 @@ def branch_allele_frequency_spectrum(
     branch_length = np.zeros(ts.num_nodes)
     tree_index = 0
 
-    def update_result(window_index, u, right):
-        if 0 < count[u, -1] < ts.num_samples:
-            x = (right - last_update[u]) * branch_length[u]
-            c = count[u, :num_sample_sets]
-            if not polarised:
-                c = fold(c, out_dim)
-            index = tuple([window_index] + list(c))
-            result[index] += x
+    def update_result(window_index, u, right, time_windows):
```

Comment:
`time_windows` isn't being changed by this function
```suggestion
    def update_result(window_index, u, right):
```